bats/foreign-keys.bats

```diff
@@ -1894,3 +1894,282 @@ SQL
     dolt add super_parent
     dolt commit -m "passes now"
 }
+
+@test "foreign-keys: Merge valid onto parent" {
+    dolt sql <<SQL
+CREATE TABLE parent (
+  pk BIGINT PRIMARY KEY,
+  v1 BIGINT
+);
+CREATE TABLE child (
+  pk BIGINT PRIMARY KEY,
+  v1 BIGINT,
+  CONSTRAINT fk_name FOREIGN KEY (v1)
+    REFERENCES parent(v1)
+);
+INSERT INTO parent VALUES (1, 1), (2, 2), (3, 3);
+INSERT INTO child VALUES (2, 1), (3, 2), (4, 3);
+SQL
+
+    dolt add -A
+    dolt commit -m "initial commit"
+    dolt checkout -b other
+    dolt checkout master
+    dolt sql <<SQL
+INSERT INTO parent VALUES (4, 3);
+SQL
+    dolt add -A
+    dolt commit -m "added row"
+    dolt checkout other
+
+    dolt sql <<SQL
+ALTER TABLE child DROP FOREIGN KEY fk_name;
+UPDATE parent SET v1 = v1 - 1;
+SQL
+    dolt add -A
+    dolt commit -m "updated parent"
+    dolt checkout master
+    dolt merge other
+    
+    run dolt sql -q "SELECT * FROM parent" -r=csv
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "pk,v1" ]] || false

```

Comment:
When you're matching a whole bunch of lines in order like this, it's more compact to echo with newlines. Like this:

`EXPECTED=$( echo -e "table,num_conflicts\none_pk,1\ntwo_pk,1")`

Reply:
Also add an order by clause so that the order doesn't vary